client/service/src/task_manager/mod.rs

```diff
@@ -48,31 +51,39 @@ pub struct SpawnTaskHandle {
 }
 
 impl SpawnTaskHandle {
-	/// Spawns the given task with the given name.
+	/// Spawns the given task with the given name and an optional group name.
+	/// If group is not specified `DEFAULT_GROUP_NAME` will be used.
 	///
-	/// Note that the `name` is a `&'static str`. The reason for this choice is that statistics
-	/// about this task are getting reported to the Prometheus endpoint (if enabled), and that
-	/// therefore the set of possible task names must be bounded.
+	/// Note that the `name`/`group` is a `&'static str`. The reason for this choice is that
+	/// statistics about this task are getting reported to the Prometheus endpoint (if enabled), and
+	/// that therefore the set of possible task names must be bounded.
 	///
 	/// In other words, it would be a bad idea for someone to do for example
 	/// `spawn(format!("{:?}", some_public_key))`.
-	pub fn spawn(&self, name: &'static str, task: impl Future<Output = ()> + Send + 'static) {
-		self.spawn_inner(name, task, TaskType::Async)
+	pub fn spawn(
+		&self,
+		name: &'static str,
+		group: Option<&'static str>,
```

Comment:
Something like:

```
enum GroupName {
    Default,
    Specific(&'static str),
}

impl From<Option<&'static str>> for GroupName {
    fn from(name: Option<&'static str>) -> Self {
        match name { Some(name) => Self::Specific(name), None => Self::Default }
    }
}

impl From<&'static str> for GroupName {
    fn from(name: &'static str) -> Self {
        Self::Specific(name)
    }
}
```

would have been super cool :P 

Because then you could write:

`pub fn spawn(.., group: impl Into<GroupName>, ..)`

And stuff like `spawn(.., "name", ..)` or `spawn(.., None, ..)` should work :P 

This would maybe be the icing on the top, but no requirement.

Reply:
Yup, you are right, that's more flexible. I'll implement this as a followup PR, at the moment I'm tackling other things.

Reply:
Sounds like an easy issue for external contributors :)

Reply:
Picking the issue up @ordian